src/hooks/useDatePicker.ts

```diff
@@ -44,14 +45,8 @@ const fromDateParts = (date: string, options: PickerOptions) => {
     let result: Temporal.PlainDateLike
 
     try {
-        const dateParts = date?.split('-')
-        if (dateParts.length !== 3) {
-            throw new Error(
-                `Invalid date ${date} - date should be in the format YYYY-MM-DD`
-            )
-        }
-        const [year, month, day] = dateParts
-        result = { year: Number(year), month: Number(month), day: Number(day) }
+        const { year, month, day } = extractDatePartsFromDateString(date)
```

Comment:
my initial idea was we don't need to expose `validateDateString` from the library - if the validation failed when trying to `extractDatePartsFromDateString` then the hook can return `invalid=false` or some error state to the consumer. The consumer doesn't need to do validation directly - wouldn't that work?

Reply:
With the current implementation, I don't need the different parts, I just need to compare the string against the `calendarDate`s in the `calendarWeekDays`. But only if the date is valid, hence exposing `validateDateString`.

Another point is that `extractDatePartsFromDateString` would throw if the date was invalid which would have to be caught and handled, while using `validateDateString` integrates better into the control flow, I'd say.